query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/QueryRequestEligibilityValidator.java

```diff
@@ -0,0 +1,154 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+import java.util.List;
+import java.util.Set;
+import org.hypertrace.core.query.service.ExecutionContext;
+import org.hypertrace.core.query.service.QueryCost;
+import org.hypertrace.core.query.service.QueryRequestUtil;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Expression.ValueCase;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.Function;
+import org.hypertrace.core.query.service.api.Operator;
+import org.hypertrace.core.query.service.api.QueryRequest;
+
+/** Set of rules to check if the given request can be served by prometheus */
+class QueryRequestEligibilityValidator {
+
+  private final PrometheusViewDefinition prometheusViewDefinition;
+
+  public QueryRequestEligibilityValidator(PrometheusViewDefinition prometheusViewDefinition) {
+    this.prometheusViewDefinition = prometheusViewDefinition;
+  }
+
+  QueryCost isEligible(QueryRequest queryRequest, ExecutionContext executionContext) {
+    // orderBy to be supported later
+    if (queryRequest.getOrderByCount() > 0) {
+      return QueryCost.UNSUPPORTED;
+    }
+
+    // only aggregation queries are supported
+    if (queryRequest.getAggregationCount() == 0
+        || queryRequest.getGroupByCount() == 0
+        || queryRequest.getDistinctSelections()) {
```

Comment:
Is distinct selection also not supported ?

Reply:
No